proof_gen/Cargo.toml

```diff
@@ -15,6 +15,7 @@ log = { workspace = true }
 paste = "1.0.14"
 plonky2 = { workspace = true }
 serde = { workspace = true }
+hashbrown = { version = "0.14.0" }
```

Comment:
We're already using `hashbrown` through `evm_arithmetization`. As such, we should move it to the workspace's `Cargo.toml` and use `hashbrown = { workspace = true }` on both underlying crates' config files.